src/plugins/basic_depth_limit.rs

```diff
@@ -28,7 +39,38 @@ impl Plugin for BasicDepthLimit {
         &self,
         service: BoxService<RouterRequest, RouterResponse, BoxError>,
     ) -> BoxService<RouterRequest, RouterResponse, BoxError> {
-        ServiceBuilder::new().service(service).boxed()
+        let limit = self.configuration.limit;
+        ServiceBuilder::new()
+            .checkpoint(move |req: RouterRequest| {
+                if let Some(operation) = req.originating_request.body().query.clone() {
+                    let ctx = ApolloCompiler::new(&operation);
+                    let operation_name = req.originating_request.body().operation_name.as_ref();
+
+                    if let Some(operation) = ctx.operation_by_name(operation_name) {
+                        let depth = operation.max_depth(&ctx);
+                        tracing::debug!("depth for operation {:?}: {}", operation_name, depth);
```

Comment:
It's generally better style with`tracing to do something like this:
```suggestion
                        tracing::debug!(%operation_name, %depth, "operation_depth");
```
This will give you a trace line which has the text field "operation_depth" and then two fields named "operation_name" and "depth" with the `Display` value for those variables. If the variables don't implement `Display`, you can use '?' instead of '%' to get a `Debug` value (if that's approporiate). If neither are available (that would be unsual) then the format you have is fine.